util/argo/ref_sources.go

```diff
@@ -0,0 +1,62 @@
+package argo
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	argoappv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/v2/util/db"
+)
+
+type GetRefSourcesOptions struct {
+	Spec       argoappv1.ApplicationSpec
+	Db         db.ArgoDB
+	Revisions  []string
+	IsRollback bool
+}
+
+// GetRefSources creates a map of ref keys (from the sources' 'ref' fields) to information about the referenced source.
+// This function also validates the references use allowed characters and does not define the same ref key more than
```

Comment:
```suggestion
// This function also validates the references use allowed characters and do not define the same ref key more than
```

Reply:
Is a spelling error on a comment a good reason to ask for a review on such a feature? Seems a good way to make contributors stop contributing.

Reply:
I took the code from argo.go and I moved it as it was and i didn't notice it. I'll update it 😄